Base/res/js/Spreadsheet/runtime.js

```diff
@@ -517,35 +511,24 @@ function internal_lookup(
     }
 
     let i = 0;
-    let didMatch = false;
-    let value = null;
     let matchingName = null;
-    lookup_inputs.forEach(name => {
-        value = thisSheet[name];
-        if (matches(value)) {
-            didMatch = true;
-            matchingName = name;
-            return Break;
-        }
-        ++i;
-    });
+    i = resolve(lookup_inputs).findIndex(val => matches(val));
```

Comment:
This pulls all input values into an array, that's definitely not desired.
e.g. a very large range will require a very large underlying array with this:
with A0: 0, A1...A1000000: 1, you'd end up allocating an array with a million entries even if the needed element was `0`.

Reply:
I think I’ll add a find then to the range class

Reply:
I see what you mean, I brought back most of the code from master, this isn't a valid concern anymore.